controllers/authController.updateProfileController.test.js

```diff
@@ -0,0 +1,234 @@
+
+        await updateProfileController(req, res);
+
+        // expect(hashPassword).toHaveBeenCalledTimes(0);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+    });
+
+    it("should keep old password if hashpassword fails", async () => {
+        hashPassword.mockReturnValueOnce(undefined);
+
+        let updatedUser = JSON.parse(JSON.stringify(newProfileUpdate));
+        updatedUser.password = "safePassword";
+
+        await updateProfileController(req, res);
+
+        // expect(hashPassword).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+    });
+
+    it("should not update empty password and empty phone fields", async () => {
+        req.body = {
+            name: sampleName,
+            email: "",
+            password: "",
+            address: validAddress,
+            phone: ""
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+        updatedUser.name = sampleName;
+        updatedUser.address = validAddress;
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    })
+
+    it("should not update empty name and empty address fields", async () => {
+        req.body = {
+            name: "",
+            email: sampleEmail,
+            password: passwordLenMoreThan6,
+            address: "",
+            phone: validPhone
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+        updatedUser.password = passwordHash;
+        updatedUser.phone = validPhone;
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    })
+
+    it("should keep old profile if new profile is empty", async () => {
+        req.body = {
+            name: "",
+            email: "",
+            password: "",
+            address: "",
+            phone: ""
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
```

Comment:
Your test is testing 'Should keep old profile if new profile is empty'. However, your `expect` statements do not seem to be verifying whether the contents of the user are being changed or not.

Reply:
I think updatedUser declared in line 182 is made from parsing the oldProfileUpdate declared in line 15. I just change the name of the updatedUser to oldProfileUpdate? Or is there some other logic error that I'm missing?